pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java

```diff
@@ -175,7 +171,22 @@ public void renderFileViolations(Iterator<RuleViolation> violations) throws IOEx
                 xmlWriter.writeCharacters(StringUtil.removedInvalidXml10Characters(rv.getDescription()));
                 writeNewLine();
                 xmlWriter.writeEndElement();
+
                 writeNewLine();
+
+                try {
+                    SourceCode sourceCode = getSourceCode(rv);
+                    if (sourceCode != null) {
+                        xmlWriter.writeStartElement("codesnippet");
```

Comment:
This should be a new renderer, as current users of the xml renderer do not expect their reports to explode in size

Reply:
Fair point. I will work on creating a new renderer with code snippet.

Reply:
Great :)